recipes/magicblast/meta.yaml

```diff
@@ -0,0 +1,66 @@
+{% set version = "1.4.0" %}
+
+package:
+  name: magicblast
+  version: {{ version }}
+
+source:
+  url: ftp://ftp.ncbi.nlm.nih.gov/blast/executables/magicblast/LATEST/ncbi-magicblast-{{ version }}-x64-linux.tar.gz # [linux]
+  url: ftp://ftp.ncbi.nlm.nih.gov/blast/executables/magicblast/LATEST/ncbi-magicblast-{{ version }}-x64-macosx.tar.gz # [macos]
+  sha256: 9995524a2ce67850516debb3341f268d485113f1c28005e39b5fbf47fe27d250 # [linux]
+  sha256: f48daeb86065be4f256820b5c189f0141eafdf50e5a0b8e837b381e82a447b0e # [macos]
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - patchelf
+  host:
+    - zlib
+    - bzip2
+    - pcre
+    - libxml2
+  run:
+    - zlib
+    - bzip2
+    - pcre
+    - libxml2
+    - blast # needed for makeblastdb
```

Comment:
Do we know which version is required?

Reply:
For blast the latest available, but which specific version is not critical. It is only needed for makeblastdb binary. For bzip2, pcre, and libxml2 I can add vesrions. How do I specify  version?

Reply:
Check the docs - can get complicated. Briefly, ` >=1.2.3,<1.3.0a0` is a good way to do it. The `,` is an `AND`, so this says "at least version 1.2.3" and "less than 1.3.0a0". The latter is the smallest possible 1.3 version. The `0a0` is because `1.3.a1 < 1.3`. Should not be strictly necessary because we don't allow alphas, but who knows what other channels a user might add. Easier than writing `1.2.999999999999999....`. 

For your later from source build: you can use `pin_compatible('libxml2', max_pin="x.x")` in the `run` section to generate dependencies based on what was installed in the `host` section. The number of `x`s determines where it pins. Given `libxml 1.2.3`, a single `x` will pin to `>=1.2.3,<2`, `x.x` to `>=1.2.3,<1.3`, while `x.x.x` will pin to `>=1.2.3,<1.2.4` (plus the `0a0` of course, it does that).

Reply:
I added library versions.